### PR TITLE
Reduce boilerplate for custom tabular bq notebook

### PR DESCRIPTION
**Pull request by ivanmkc**
- Reduce boilerplate for custom tabular bq notebook
- Default region to us-central1 to reduce user friction.

### CODE REVIEW

File: .cloud-build/execute_changed_notebooks_helper.py [hunk 1]
```diff
@@ -111,6 +111,7 @@ def _process_notebook(
     ) = remove_no_execute_cells_preprocessor.preprocess(nb)
 
     (nb, resources) = update_variables_preprocessor.preprocess(nb, resources)
+    (nb, resources) = unique_strings_preprocessor.preprocess(nb, resources)
```
Comment (ivanmkc) [inline .cloud-build/execute_changed_notebooks_helper.py:114]: This was missing before.

File: .cloud-build/utils/NotebookProcessors.py [hunk 2]
```diff
@@ -73,13 +73,19 @@ def generate_uuid(length: int = 8) -> str:
...
     @staticmethod
     def update_unique_strings(content: str):
-        # Replace strings that end with "-unique" with a uuid.
-
-        return content.replace('-unique"', f'-{generate_uuid()}"')
+        # Replace strings that end with "-unique" or "_unique" with a uuid.
+
+        unique_id = generate_uuid()
+        return (
+            content.replace('-unique"', f'-{unique_id}"')
```
Comment (ivanmkc) [inline .cloud-build/utils/NotebookProcessors.py:84]: Added more variations.

File: .cloud-build/utils/util.py [hunk 3]
```diff
@@ -61,9 +61,7 @@ def archive_code_and_upload(staging_bucket: str):
 
 
 def download_blob_into_memory(
-  bucket_name: str,
-  blob_name: str,
-  download_as_text: Optional[bool]=False
+    bucket_name: str, blob_name: str, download_as_text: Optional[bool] = False
```
Comment (ivanmkc) [inline .cloud-build/utils/util.py:64]: Lint.

File: .cloud-build/utils/util.py [hunk 4]
```diff
@@ -79,13 +77,10 @@ def download_blob_into_memory(
 
     # Download the blob content
     if download_as_text:
-      contents = blob.download_as_text()
+        contents = blob.download_as_text()
```
Comment (ivanmkc) [inline .cloud-build/utils/util.py:80]: Lint.

**Review by andrewferlitsch — APPROVED** [review]

File: notebooks/official/custom/custom-tabular-bq-managed-dataset.ipynb [hunk 5]
```diff
@@ -123,55 +123,6 @@
         "to generate a cost estimate based on your projected usage."
```
Comment (ivanmkc) [inline notebooks/official/custom/custom-tabular-bq-managed-dataset.ipynb:123]: <p>I will add this notebook tomorrow, I won't merge this PR until that is in.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1207/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

File: notebooks/notebook_template.ipynb [hunk 6]
```diff
@@ -255,7 +255,7 @@
       },
```
Comment (ivanmkc) [inline notebooks/notebook_template.ipynb:255]: <p>Redirect to detailed doc regarding service account permissions.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1207/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Pull request merged**